src/robot/output/listenerarguments.py

```diff
@@ -72,7 +72,8 @@ def _get_version2_arguments(self, item):
         attributes = dict((name, self._get_attribute_value(item, name))
                           for name in self._attribute_names)
         attributes.update(self._get_extra_attributes(item))
-        return item.name, attributes
+        name = getattr(item, 'unresolved_name', None) or item.name
```

Comment:
The real name should be passed here instead of conditionally checking what name to use.

Also, if I understand the logic correctly, `name` and `unresolved_name` end up being exactly the same with tests. That wouldn't be useful at all. Having `name` with the original name and separate `resolved_name` would potentially be useful, but I think `resolved_name` can be added later if needed.

Reply:
initially, i haven't this logic. The argument `name`  of listener v2 and the `result.name` of listener v3 had the resolved test cases name.

But we want the argument `name`  of listener v2 have the unresolved test cases name and the `result.name` of listener v3 had the resolved test cases name and i would like keep the same logic.

A other solution is let the name of the result unresolved, add attribute `resolved_name` on model of test case result and just change in `robot/output/xmllogger.py` the printed value by `start_test`.

Reply:
Do you mean using `ModelCombiner(test, result, name=test.name)` like I proposed below breaks listener v3 or doesn't work otherwise?

Reply:
Your proposition works, it is so simple. I will apply change tomorrow.